### PR TITLE
[Vertical compaction 3/5] support partial columns write in segment writer

### PR DESCRIPTION
**Pull request by wyb**
#1112 

Add partical columns write interface in segment writer.

### CODE REVIEW

**Comment by wyb** [comment]
run starrocks_be_unittest

File: be/src/storage/rowset/segment_v2/segment_writer.cpp [hunk 1]
```diff
@@ -66,20 +66,39 @@ void SegmentWriter::_init_column_meta(ColumnMetaPB* meta, uint32_t* column_id, c
     }
 }
 
-Status SegmentWriter::init(uint32_t write_mbytes_per_sec __attribute__((unused))) {
-    uint32_t column_id = 0;
+Status SegmentWriter::init() {
+    std::vector<uint32_t> all_column_indexes;
```
Comment (sduzh) [inline be/src/storage/rowset/segment_v2/segment_writer.cpp:70]: Better to reserve it before the emplace

File: be/src/storage/rowset/segment_v2/segment_writer.cpp [hunk 2]
```diff
@@ -153,71 +176,72 @@ uint64_t SegmentWriter::estimate_segment_size() {
...
+            return Status::InternalError(
+                    strings::Substitute("column index $0 out of range $1", column_index, num_columns));
+        }
 
-// write ordinal index after data has been written
-Status SegmentWriter::_write_ordinal_index() {
-    for (auto& column_writer : _column_writers) {
+        auto& column_writer = _column_writers[i];
+        RETURN_IF_ERROR(column_writer->finish());
+        // write data
```
Comment (sduzh) [inline be/src/storage/rowset/segment_v2/segment_writer.cpp:202]: Meaningless comment, the method itself is self-explained

**Review by sduzh — DISMISSED** [review]

File: be/src/storage/rowset/segment_v2/segment_writer.cpp [hunk 3]
```diff
@@ -52,9 +52,9 @@ SegmentWriter::SegmentWriter(std::unique_ptr<fs::WritableBlock> wblock, uint32_t
 
 SegmentWriter::~SegmentWriter() {}
 
-void SegmentWriter::_init_column_meta(ColumnMetaPB* meta, uint32_t* column_id, const TabletColumn& column) {
+void SegmentWriter::_init_column_meta(ColumnMetaPB* meta, uint32_t column_id, const TabletColumn& column) {
     // TODO(zc): Do we need this column_id??
```
Comment (chaoyli) [inline be/src/storage/rowset/segment_v2/segment_writer.cpp:56]: This TODO can be removed.

**Comment by chaoyli** [comment]
It's better to give different commit message about different pull request. @wyb

**Review by chaoyli — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Pull request merged**